lib/middleware/request-handler.ts

```diff
@@ -265,6 +265,12 @@ export function onSubscribe(req: Interface.IOurRequest,
                 // Buffer the current data
                 sub.buffer.pushValue(data);
             });
+
```

Comment:
It would be better to add a comments about why we need this event listener, and specifically pointing out the it doesn't need to **unsubscribe** when `error` occurs.